sdk2013: Mac: debug: issues with memdbgon.h

**To reproduce:**
1. Clone Metamod:Source
1. Clone the `sdk2013` branch
1. Build sample_mm from Metamod:Source with the `sdk2013` SDK

**Expected:**
Build without error.

**Actual:*
Fails with error:
```
hl2sdk-sdk2013/public/tier0/memdbgon.h:39:10: fatal error: 'malloc.h' file not found
#include <malloc.h>
         ^~~~~~~~~~
1 error generated.
```

If that is fixed (see Possible Fix), fails with the following:
```
hl2sdk-sdk2013/public/tier1/utlstring.h:19:17: error: expected parameter declarator
inline wchar_t *wcsdup(const wchar_t *pString)
                ^
hl2sdk-sdk2013/public/tier0/memdbgon.h:132:41: note: expanded from macro 'wcsdup'
#define wcsdup(s)  MemAlloc_WcStrDup(s, __FILE__, __LINE__)
                                        ^
<scratch space>:30:1: note: expanded from here
"hl2sdk-sdk2013/public/tier1/utlstring.h"
^

hl2sdk-sdk2013/public/tier1/utlstring.h:19:17: error: expected ')'
hl2sdk-sdk2013/public/tier0/memdbgon.h:132:41: note: expanded from macro 'wcsdup'
#define wcsdup(s)  MemAlloc_WcStrDup(s, __FILE__, __LINE__)
                                        ^
<scratch space>:30:1: note: expanded from here
"hl2sdk-sdk2013/public/tier1/utlstring.h"
^
hl2sdk-sdk2013/public/tier1/utlstring.h:19:17: note: to match this '('
hl2sdk-sdk2013/public/tier0/memdbgon.h:132:37: note: expanded from macro 'wcsdup'
#define wcsdup(s)  MemAlloc_WcStrDup(s, __FILE__, __LINE__)
```

**Possible fix:**
In platform.h there is the following:
```C++
#if defined __APPLE__
#include <stdlib.h>
#else
#include <malloc.h>
#endif
```

Seems like this is the canonical way to overcome the missing `malloc.h` header. Another options is to include `<malloc/malloc.h>`, but best to just use the same approach as in `platform.h`

- - -

The other bug happens because both `tier1/utlstring.h` and `tier0/memdbgon.h` try to provide `wcsdup` on Mac debug builds.

A potential fix is to conditionally disable the implementation in `utlstring.h` to let `memdbg` track allocations, as the implementations only differ in that `memdbg` calls `g_pMemAlloc->Alloc` and not `malloc` and tracks `__FILE__` and `__LINE__`.

--- a/public/tier0/memdbgon.h
+++ b/public/tier0/memdbgon.h
@@ -33,7 +33,11 @@
 #include <wchar.h>
 #endif
 #include <string.h>
+#if defined __APPLE__
+#include <stdlib.h>
+#else
 #include <malloc.h>
+#endif
 #include "commonmacros.h"
 #include "memalloc.h"
 

--- a/public/tier1/utlstring.h
+++ b/public/tier1/utlstring.h
@@ -16,6 +16,8 @@
 #include "limits.h"
 
 #if defined( OSX )
+
+#if !defined( wcsdup )
 inline wchar_t *wcsdup(const wchar_t *pString)
 {
 	wchar_t *pMemory;
@@ -31,6 +33,7 @@ inline wchar_t *wcsdup(const wchar_t *pString)
 
 	return NULL;
 }
+#endif
 
 inline size_t strnlen(const char *s, size_t n)
 {

